core/src/main/java/org/mule/TransformationService.java

```diff
@@ -0,0 +1,486 @@
+     * @param outputType the desired return type
+     * @return The converted payload of this message. Note that this method will not alter the payload of this
+     * message *unless* the payload is an InputStream in which case the stream will be read and the payload will become
+     * the fully read stream.
+     * @throws TransformerException if a transformer cannot be found or there is an error during transformation of the
+     * payload
+     */
+    public <T> T getPayload(MuleMessage message, Class<T> outputType) throws TransformerException
+    {
+        return getPayload(message, DataTypeFactory.create(outputType), message.getEncoding());
+    }
+
+    /**
+     * Will attempt to obtain the payload of this message with the desired Class type. This will
+     * try and resolve a transformer that can do this transformation. If a transformer cannot be found
+     * an exception is thrown.  Any transformers added to the registry will be checked for compatability
+     * @param outputType the desired return type
+     * @return The converted payload of this message. Note that this method will not alter the payload of this
+     * message *unless* the payload is an InputStream in which case the stream will be read and the payload will become
+     * the fully read stream.
+     * @throws TransformerException if a transformer cannot be found or there is an error during transformation of the
+     * payload
+     */
+    public <T> T getPayload(MuleMessage message, DataType<T> outputType) throws TransformerException
+    {
+        return getPayload(message, outputType, message.getEncoding());
+    }
+
+    /**
+     * Converts the message implementation into a String representation. If encoding
+     * is required it will use the encoding set on the message
+     *
+     * @return String representation of the message payload
+     * @throws Exception Implementation may throw an endpoint specific exception
+     *
+     */
+    public final String getPayloadAsString(MuleMessage message) throws Exception
+    {
+        return getPayloadAsString(message, message.getEncoding());
+    }
+
+    /**
+     * Get the message payload for logging without throwing exception
+     * Converts the message implementation into a String representation. If encoding
+     * is required it will use the encoding set on the message
+     *
+     * @return message payload as object
+     */
+    public String getPayloadForLogging(MuleMessage message, String encoding)
+    {
+        try
+        {
+            return getPayloadAsString(message, encoding);
+        }
+        catch (Exception e)
+        {
+            return  "[Message could not be converted to string]";
+        }
+    }
+
+    /**
+     * Get the message payload for logging without throwing exception
+     * Converts the message implementation into a String representation.
+     *
+     * @return message payload as object
+     */
+    public String getPayloadForLogging(MuleMessage message)
```

Comment:
All these methods should clarify that the message payload could be replaced with the transformed value when the payload is consumable and that the message could be in an inconsistent state if the paylod was consume but the transformed value is not of the expected type